src/lib.rs

```diff
@@ -1,808 +1,268 @@
-                        let mut sess = ssh::Session::connect(
-                            &log,
-                            &usernames[name],
-                            SocketAddr::new(
-                                machine
-                                    .public_ip
-                                    .parse::<IpAddr>()
-                                    .context("machine ip is not an ip address")?,
-                                22,
-                            ),
-                            private_key_file.path(),
-                            wait_for,
-                        ).context(format!(
-                            "failed to ssh to {} machine {}",
-                            name, machine.public_dns
-                        ))
-                            .map_err(|e| {
-                                error!(log, "failed to ssh to {}:{}", &name, &machine.public_ip);
-                                e
-                            })?;
-
-                        debug!(log, "setting up {} instance", name; "ip" => &machine.public_ip);
-                        f(&mut sess)
-                            .context(format!("setup procedure for {} machine failed", name))
-                            .map_err(|e| {
-                                error!(log, "setup for {} machine failed", name);
-                                e
-                            })?;
-                        info!(log, "finished setting up {} instance", name; "ip" => &machine.public_ip);
-
-                        machine.ssh = Some(sess);
-                        Ok(())
-                    })
-                    .filter_map(Result::err)
-            }));
-
-            if errors.is_empty() {
-                // 4. invoke F with Machine descriptors
-                let start = time::Instant::now();
-                info!(log, "quiet before the storm");
-                res = Some(
-                    f(machines)
-                        .context("tsunami main routine failed")
-                        .map_err(|e| {
-                            crit!(log, "main tsunami routine failed");
-                            e
-                        })?,
-                );
-                info!(log, "the power of the tsunami was unleashed"; "duration" => start.elapsed().as_secs());
-            }
-        } else {
-            crit!(
-                log,
-                "only {} out of {} machines were started; aborting",
-                running,
-                expected_num
-            );
-        }
-
-        debug!(log, "all done");
+        Ok(())
+    }
+}
 
-        // TODO: this will only expose first setup error -- fix that
-        errors
-            .into_iter()
-            .next()
-            .map(|e| Err(e))
-            .unwrap_or_else(|| {
-                Ok(res.expect("if there are no errors, then we ran the user's main function"))
-            })
+#[cfg(test)]
+mod test {
+    pub(crate) fn test_logger() -> slog::Logger {
```

Comment:
What does this test really test?

Reply:
it's a helper method for other tests. Is it supposed to go somewhere else?

Reply:
Ohhh, I see. I missed that it wasn't `#[test]`. Never mind.